TODO_client_hardening

```diff
@@ -0,0 +1,10 @@
+* time requests and log long running ones
+* handle Errno::EPIPE.
+  Maybe use httpclient instead ?
+* handle DU issues
+
+ideas
+* parametrize default timeout ?
+* make default timeout client specific ?
```

Comment:
Before merging this needs to be moved out into separate GitHub issues. I can add a `stability` tag for you :+1: